src/renderable/sprite.js

```diff
@@ -262,12 +262,19 @@
             var w = this.width, h = this.height;
             var angle = this.angle + this._sourceAngle;
 
+            // save context
+            renderer.save();
+            
+            // calculate pixel pos of the anchor point
+            var ax = w * this.anchorPoint.x, ay = h * this.anchorPoint.y;
+            renderer.translate(-ax, -ay);
+            
+            // adjust geometry box position based on anchor point offset
+            this._leftOffset = xpos - ax;
+            this._topOffset = ypos - ay;
```

Comment:
Less is more: https://gist.github.com/parasyte/fa9664cc19824c4ff7b7

I added the same translation into the debugPanel, so the conditional geometry offsets can be removed. The translation is exactly the same as what you do here. It's identical to the method used to draw the hitboxes around the entity; debugPanel needs to do the same translation for those hitboxes as the entity does for its child renderable.